lib/creatubbles/base_object.rb

```diff
@@ -9,7 +9,16 @@ def self.type_name
     @type_name
   end
 
+  def self.attributes
+    @attributes || []
+  end
+
+  def self.relationships
+    @relationships || []
+  end
+
   def self.define_attributes(atts)
+    @attributes = atts
```

Comment:
some general considerations: when saving a method parameter as instance variable, it's very much possible that 
the caller of the method keeps a reference to that parameter (atts) and manipulates it afterwards. This will then also affect `@attributes`, as it's same object.
Further, if you'd end up calling `define_attributes` twice, the second invocation will overwrite the `@attributes`.
So better to have the `@attributes` array be owned by this object, and have `define_attributes` add new attributes to that array. Also, should normalise the entries there, i.e. make sure all entries are symbols.